### PR TITLE
Issue #7639: update doc for ExplicitInitialization

### PR DESCRIPTION
**Pull request by ImmortalRabbit**
Issue #7639: update doc for ExplicitInitialization

image
image

Output of default example:
$ cat config.xml
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
      <module name="ExplicitInitialization"/>
    </module>
</module>
```
$ cat Test.java
```
public class Test {
    private int a = 0; // violation
    private int b = 1;
    private int c = 2;

    private char a = '\0'; // violation
    private char b = 'b';
    private char c = 'c';

    private boolean a = true;
    private boolean b = false; // violation
    private boolean c = true;
    private boolean d = false; // violation
    private boolean e = false; // violation

    private A a = new A();
    private A b = null; // violation
    private C c = null; // violation
    private D d = new D();

    int ar1[] = null; // violation
    int ar2[] = new int[10];
    int ar3[];

    private Bar<String> bar = null; // violation
    private Bar<String>[] barArray = null; // violation
}
```

$ java -jar checkstyle/target/checkstyle-8.31-SNAPSHOT-all.jar -c config.xml java_sandbox/src/com/codestyle/sandbox/Test.java

```
Starting audit...
[ERROR] /Users/demezhanmarikov/IdeaProjects/java_sandbox/src/com/codestyle/sandbox/Test.java:4:17: Variable 'a' explicitly initialized to '0' (default value for its type). [ExplicitInitialization]
[ERROR] /Users/demezhanmarikov/IdeaProjects/java_sandbox/src/com/codestyle/sandbox/Test.java:8:18: Variable 'a' explicitly initialized to '\0' (default value for its type). [ExplicitInitialization]
[ERROR] /Users/demezhanmarikov/IdeaProjects/java_sandbox/src/com/codestyle/sandbox/Test.java:13:21: Variable 'b' explicitly initialized to 'false' (default value for its type). [ExplicitInitialization]
[ERROR] /Users/demezhanmarikov/IdeaProjects/java_sandbox/src/com/codestyle/sandbox/Test.java:15:21: Variable 'd' explicitly initialized to 'false' (default value for its type). [ExplicitInitialization]
[ERROR] /Users/demezhanmarikov/IdeaProjects/java_sandbox/src/com/codestyle/sandbox/Test.java:16:21: Variable 'e' explicitly initialized to 'false' (default value for its type). [ExplicitInitialization]
[ERROR] /Users/demezhanmarikov/IdeaProjects/java_sandbox/src/com/codestyle/sandbox/Test.java:19:15: Variable 'b' explicitly initialized to 'null' (default value for its type). [ExplicitInitialization]
[ERROR] /Users/demezhanmarikov/IdeaProjects/java_sandbox/src/com/codestyle/sandbox/Test.java:20:15: Variable 'c' explicitly initialized to 'null' (default value for its type). [ExplicitInitialization]
[ERROR] /Users/demezhanmarikov/IdeaProjects/java_sandbox/src/com/codestyle/sandbox/Test.java:23:9: Variable 'ar1' explicitly initialized to 'null' (default value for its type). [ExplicitInitialization]
[ERROR] /Users/demezhanmarikov/IdeaProjects/java_sandbox/src/com/codestyle/sandbox/Test.java:27:25: Variable 'bar' explicitly initialized to 'null' (default value for its type). [ExplicitInitialization]
[ERROR] /Users/demezhanmarikov/IdeaProjects/java_sandbox/src/com/codestyle/sandbox/Test.java:28:27: Variable 'barArray' explicitly initialized to 'null' (default value for its type). [ExplicitInitialization]
Audit done.

```
Output of default example:
$ cat config.xml
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="ExplicitInitialization">
           <property name="onlyObjectReferences" value="true"/>
        </module>
     </module>
</module>
```
$ cat Test.java
```
public class Test {
    private int a = 0;
    private int b = 1;
    private int c = 2;

    private char a = '\0';
    private char b = 'b';
    private char c = 'c';

    private boolean a = true;
    private boolean b = false;
    private boolean c = true;
    private boolean d = false;
    private boolean e = false;

    private A a = new A();
    private A b = null; // violation
    private C c = null; // violation
    private D d = new D();

    int ar1[] = null; // violation
    int ar2[] = new int[10];
    int ar3[];

    private Bar<String> bar = null; // violation
    private Bar<String>[] barArray = null; // violation
}
```

$ java -jar checkstyle/target/checkstyle-8.31-SNAPSHOT-all.jar -c config.xml java_sandbox/src/com/codestyle/sandbox/Test.java

```
Starting audit...
[ERROR] /Users/demezhanmarikov/IdeaProjects/java_sandbox/src/com/codestyle/sandbox/Test.java:19:15: Variable 'b' explicitly initialized to 'null' (default value for its type). [ExplicitInitialization]
[ERROR] /Users/demezhanmarikov/IdeaProjects/java_sandbox/src/com/codestyle/sandbox/Test.java:20:15: Variable 'c' explicitly initialized to 'null' (default value for its type). [ExplicitInitialization]
[ERROR] /Users/demezhanmarikov/IdeaProjects/java_sandbox/src/com/codestyle/sandbox/Test.java:23:9: Variable 'ar1' explicitly initialized to 'null' (default value for its type). [ExplicitInitialization]
[ERROR] /Users/demezhanmarikov/IdeaProjects/java_sandbox/src/com/codestyle/sandbox/Test.java:27:25: Variable 'bar' explicitly initialized to 'null' (default value for its type). [ExplicitInitialization]
[ERROR] /Users/demezhanmarikov/IdeaProjects/java_sandbox/src/com/codestyle/sandbox/Test.java:28:27: Variable 'barArray' explicitly initialized to 'null' (default value for its type). [ExplicitInitialization]
Audit done.

```

### CODE REVIEW

**Comment by strkkk** [comment]
@ImmortalRabbit please update first post with site screenshot and checkstyle run output for examples.
For example https://github.com/checkstyle/checkstyle/pull/7789

**Comment by rdiachenko** [comment]
@ImmortalRabbit any update on this PR?

**Comment by ImmortalRabbit** [comment]
@rdiachenko Sorry, I was busy with my classes, gonna do fixes tomorrow. I hope thats okay.

**Comment by ImmortalRabbit** [comment]
> @ImmortalRabbit please update first post with site screenshot and checkstyle run output for examples.
> For example #7789 
Sorry could help me? I couldn't find how I can view site with my changes to make screenshots of my changes. 
Thanks a lot.

**Comment by DXTkastb** [comment]
@ImmortalRabbit 
https://github.com/checkstyle/checkstyle/wiki/How-to-run-certain-phases-and-validations#how-to-generate-website-only.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ExplicitInitializationCheck.java [hunk 1]
```diff
@@ -67,15 +67,19 @@
  * </p>
  * <pre>
  * public class Test {
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/ExplicitInitializationCheck.java:69]: please add example of code below default config to show users how it works by default.
The same code should be used in config with property to let user see difference in behavior.

**Pull request closed without merging**